### ISSUE TITLE
Config max size for images uploaded/pasted/dragged-in

### ISSUE THREAD

**Issue by SoorajSethumadhavan** [tinymce/tinymce#4897]
This is a feature request.

Can we have an optional configuration to have maxImageFileSize, so that on trying to paste/upload/drag-and-drop images we have a validation checking if image size exceeds maxImageFileSize.

If it exceeds, then an error 'balloon' of the notificationManager can show up and say that 'Max upload size is ___' or something of that sort.. OR, we can have callback to let the user decide what to do.

**Comment by SoorajSethumadhavan** [comment]
I believe changes are needed in paste plugin?? Ignore me if I am wrong.. :)

**Comment by TheSpyder** [comment]
My preferred solution to this uses image dimensions rather than file size. The two aren't always linked but it's a fairly safe bet that if an image is resampled down within configurable width and height bounds it will reduce the image file size considerably.

For scenarios where the `imagetools` plugin isn't available, then yes we would fall back to rejecting the image with a notification banner.

**Comment by vishwakarmanikhil** [comment]
There is no solution for this till now. I am exactly looking for the same thing. Any solution?

**Comment by Vaiterius** [comment]
Yes, this is a great and very useful idea. Please implement this.

**Comment by samyouel** [comment]
Yes please!!!

**Comment by LGinC** [comment]
```js

tinymce.init({
  selector: 'textarea', 
  plugins: 'image',
  toolbar: 'image',
  images_file_types: 'svg,webp',
  images_upload_handler: image_upload_handler 
});


const image_upload_handler = (blobInfo, progress) => new Promise((resolve, reject) => {
 if (blobInfo.blob().size > 102400) {
        reject('can not  over 100KB');
        return;
    }

//upload process code .....
}
```

it work on v7, but so complex, we need `maxImageFileSize: '100KB'`
